### DISCUSSION TITLE
scalar comparisions treat BOOL modifier as optional

### DISCUSSION THREAD

**Discussion by yuanbohan** [GreptimeTeam/promql-parser#59] [Ideas]
`2 > 1` and `2 > BOOL 1` both should be valid expressions

**Comment by waynexia** [comment]
Agreed. Even in the PromQL context, `2>1` should also be executable as other expressions. The behavior may look the same with `2 > BOOL 1`, which returns 1 for true and 0 for false.

From my perspective, default comparison expression's behavior to filter rather than return values is not intuitive. I can hardly recall another system/language that has this behavior...
